internal/resources/common/scope/scope_schema.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package commonscope
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"golang.org/x/exp/slices"
+)
+
+type (
+	Scope int64
+
+	SchemaOption func(*SchemaConfig)
+
+	SchemaConfig struct {
+		Description string
+		Scopes      []string
+	}
+)
```

Comment:
Don't we already have this common scope?

Reply:
yes we do have, but in my fork, I don't have feat/pde branch so that's why I have to add it again
while merging I'll take care for this.

Reply:
Thanks, please create an issue to track this.